lib/puppet/type/dism.rb

```diff
@@ -52,15 +40,4 @@
     #defaultto([0, 3010])
     defaultto([0, 3010, 3010 & 0xFF])
   end
-
-  def munge_boolean(value)
-    case value
-      when true, "true", :true
-        :true
-      when false, "false", :false
-        :false
-      else
-        fail("munge_boolean only takes booleans")
-    end
-  end
```

Comment:
The `newvalues(:true,:false)` mean that only those two values would be allowed. I think the issue lay in the fact that the module internally mixed different kinds of parameters, and similarly the spec tests did.